api/pacman-api-admin/src/main/java/com/tmobile/pacman/api/admin/repository/service/plugins/AbstractPluginService.java

```diff
@@ -369,4 +381,23 @@ protected void updateAssetGroupStatusForOptionalAssetType(PluginParameters param
             LOGGER.error("Could not activity log for plugin {} with account id {}", parameters.getPluginName(), parameters.getId());
         }
     }
+
+    protected JsonNode getApiResponse(HttpUriRequest request) {
+        try (CloseableHttpClient httpClient = HttpClients.createDefault()) {
+            try (CloseableHttpResponse response = httpClient.execute(request)) {
+                int statusCode = response.getStatusLine().getStatusCode();
+                if (statusCode == 200) {
+                    return objectMapper.readTree(response.getEntity().getContent());
+                } else {
+                    throw new PluginApiResponseException("HTTP request failed", statusCode);
+                }
+            }
+        } catch (UnknownHostException e) {
+            LOGGER.error("Unknown host error: " + e.getMessage(), e);
+            throw new PluginApiResponseException("Request failed due to Unknown Host", 1000, e);
```

Comment:
again, stop throwing exceptions everywhere. It's not the way to handle situation. you have to do error handling, not error producing. 
if method has two states of execution - success or fail - make it return corresponding data structure.

Reply:
updated